### DISCUSSION TITLE
Feature Request: getCrudTransactionBatch for uploading data

### DISCUSSION THREAD

**Discussion by ImSingee** [powersync-ja/powersync-js#415] [General]
The current `getCrudBatch` does not care about transactions, this can break transactions in some edge casts.

I expect a `getNextCrudTransactionsBatch` method, which can return more than one transaction at a time.

What I expect I can do:

1. Get a batch of transactions to process
2. Upload transactions to the server, and complete all transactions or first few transactions

**Comment by stevensJourney** [comment]
I can see how that could be useful, but I imagine exposing a mechanism for completing only a subset of transactions could get tricky.

In the interim it should already be possible to fetch multiple `CrudTransaction`s by calling `getNextCrudTransaction` in a loop. Granular control of completing transactions could be managed by maintaining a reference to each `CrudTransaction`s `complete` method.

**Comment by ImSingee** [comment]
@stevensJourney Call `getNextCrudTransaction` multiple times without calling `complete` will return the same transaction.

Yes, it shouldn't allow completing an arbitrary subset of transactions, only the first X transactions should be allowed to be completed.

**Comment by stevensJourney** [comment]
Ah, you are correct. I overlooked that behaviour. 

I'll see that this request is added to our internal tracker.
